test/e2e/ratelimit/ratelimit_universal.go

```diff
@@ -117,9 +120,12 @@ conf:
 		err := YamlUniversal(specificRateLimitPolicy)(cluster)
 		Expect(err).ToNot(HaveOccurred())
 
-		time.Sleep(15 * time.Second)
+		Eventually(func() bool {
+			_, _, err := cluster.Exec("", "", "demo-client", "curl", "-v", "--fail", "echo-server_kuma-test_svc_8080.mesh")
+			return err == nil
+		}, "30s", "3s").Should(BeTrue())
```

Comment:
You cannot assume that this policy is automatically applied in 0ms latency. I think this test will be flaky.

Reply:
That is the point of Eventually, to make sure that in 30s we'll get at least one pass.

Reply:
I might misread the code, but I think the flow right now is like this with the second `It()`
* we apply `rateLimitPolicy` in BeforeSuite
* You apply new policy
* We now have this eventually block that immediately will pass regardless if new policy is propagated to envoys or not
* You go ahead and do the X requests. It might be the case that those requests are still using the old policy